04.ls/ls_command.rb

```diff
@@ -43,4 +67,38 @@ def display_files(files)
   end
 end
 
+def display_file_details(files)
+  file_stats = files.to_h { |file| [file, File::Stat.new(file)] }
+
+  puts "total #{file_stats.values.sum(&:blocks)}"
+  print find_file_details(file_stats).join("\n")
+end
+
+def find_file_details(file_stats)
+  linksize_width = file_stats.values.map(&:nlink).max.to_s.bytesize + 2
+  filesize_width = file_stats.values.map(&:size).max.to_s.bytesize + 2
+
+  file_stats.map do |file, file_stat|
+    cols = []
+    cols << FILE_TYPE_CHARACTER[file_stat.ftype]
+    cols << convert_file_permissions(file_stat)
+    cols << file_stat.nlink.to_s.rjust(linksize_width)
+    cols << " #{Etc.getpwuid(file_stat.uid).name}  #{Etc.getgrgid(file_stat.gid).name}"
+    cols << file_stat.size.to_s.rjust(filesize_width)
+    cols << find_updated_time(file_stat)
```

Comment:
ここもfindはちょっと変ですね。
ここでやってるのは日付データの整形だと思うので、
```suggestion
    cols << format_updated_time(file_stat)
```

みたいな名前が良いかなと思います